src/robotFactory.js

```diff
@@ -1,10 +1,96 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    if (step > 0) {
+      this.coords.y += step;
+    }
```

Comment:
```suggestion
    this.coords.y += step;
```
it seems no need for this validation))